sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -498,21 +473,13 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
       throw new TypeError("The parameter 'onError' must be of type 'function'.");
     }
 
-    this._createMessageSessionIfDoesntExist()
-      .then(async () => {
-        if (!this._messageSession) {
-          return;
-        }
-        if (!this._isClosed) {
-          this._messageSession.receive(onMessage, onError, options);
-        } else {
-          await this._messageSession.close();
-        }
-        return;
-      })
-      .catch((err) => {
+    if (!this._isClosed) {
```

Comment:
`_throwIfReceiverOrConnectionClosed()` would have handled the case where `this._isClosed` is true, so we dont need this check.

Reply:
Nice catch, updated.